src/Drips.sol

```diff
@@ -501,9 +501,10 @@ library Drips {
                     newDefaultEnd
                 );
                 _setDeltaRange(state.amtDeltas, cycleSecs, start, end, newRecv.amtPerSec);
-                // The receiver may have never been used, initialize it
-                if (state.nextCollectedCycle == 0) {
-                    state.nextCollectedCycle = _currTimestamp() / cycleSecs + 1;
+                // Ensure that the receiver collects the updated cycles
+                uint32 startCycle = start / cycleSecs + 1;
+                if (state.nextCollectedCycle == 0 || state.nextCollectedCycle > startCycle) {
+                    state.nextCollectedCycle = startCycle;
```

Comment:
Uh. That was actually a bug. These scheduled drips.
We should write a test for it.

Something like. 
First add:
Drip 1: starts at timestamp 10

Second add:
Drip 2: starts at timestamp 5

Reply:
It wasn't a bug, `state.nextCollectedCycle > startCycle` would never happen, because `nextCollectedCycle` was initialized to `_currTimestamp() / cycleSecs + 1`. `startCycle` is capped to never go below `_currTimestamp() / cycleSecs + 1`.

Reply:
Ah okay. I somehow thought you changed that.

Would it make sense to change it? Basically, If I want to schedule a drip 10 months into the future. If it is the only drip of the receiver, we wouldn't need to iterate the first 10 months.

Reply:
I'm a little confused about your comment, haven't you just described this exact PR?

Reply:
All good. I got confused I somehow thought we set the `nextCollectedCycle` somewhere else and in my mind I only considered the update cases here.

Reply:
We forgot the update in the `pickCurr && pickNew` case. We should open a separate PR to fix it.